compiler/rustc_data_structures/src/intern.rs

```diff
@@ -108,5 +110,76 @@ where
     }
 }
 
+/// A helper trait so that `Interned` things can cache stable hashes reproducibly.
+pub trait InternedHashingContext {
+    fn with_def_path_and_no_spans(&mut self, f: impl FnOnce(&mut Self));
+}
+
+#[derive(Copy, Clone)]
+pub struct InTy<T> {
```

Comment:
What's the meaning of the name `InTy`? It's not obviously meaningful to me.

Reply:
Feels like there should be a comment on this type, too.

Reply:
Also, is this `InTy` wrapping likely to occur with other types? The generic `T` here suggests yes.

Reply:
I documented and renamed this.